e2e/system_test.go

```diff
@@ -199,3 +200,182 @@ func TestSystem_StakeAmount(t *testing.T) {
 		})
 	}
 }
+
+func TestSystem_UnstakeAmount(t *testing.T) {
+	addressKeyPairs := generateAddressKeyPairs(2, t)
+
+	signer := crypto.NewEIP155Signer(100)
+
+	stakingAddress := types.StringToAddress(system.GetOperationsMap()["staking"])
+	unstakingAddress := types.StringToAddress(system.GetOperationsMap()["unstaking"])
+
+	validAccounts := []struct {
+		address       types.Address
+		balance       *big.Int
+		stakedBalance *big.Int
+	}{
+		// Staking address initialization
+		{
+			stakingAddress,
+			framework.EthToWei(10), // 10 ETH has been staked in the past
+			framework.EthToWei(0),
+		},
+		// Valid account with stake
+		{
+			addressKeyPairs[0].address,
+			framework.EthToWei(50), // 50 ETH
+			framework.EthToWei(10), // 10 ETH
+		},
+		// Valid account without stake
+		{
+			addressKeyPairs[1].address,
+			big.NewInt(0),
+			framework.EthToWei(0),
+		},
+	}
+
+	testTable := []struct {
+		name          string
+		staker        types.Address
+		unstakeAmount *big.Int
+		shouldSucceed bool
+	}{
+		{
+			"Valid unstake",
+			validAccounts[1].address,
+			framework.EthToWei(10),
+			true,
+		},
+		{
+			"Invalid unstake",
+			validAccounts[2].address,
+			framework.EthToWei(100),
+			false,
+		},
+	}
+
+	srvs := framework.NewTestServers(t, 1, func(config *framework.TestServerConfig) {
+		config.SetConsensus(framework.ConsensusDev)
+		config.SetSeal(true)
+
+		for _, acc := range validAccounts {
+			config.PremineWithStake(acc.address, acc.balance, acc.stakedBalance)
+		}
+	})
+	srv := srvs[0]
+
+	rpcClient := srv.JSONRPC()
+	for indx, testCase := range testTable {
+		t.Run(testCase.name, func(t *testing.T) {
+			// Fetch the staker balance before sending the transaction
+			accountBalance, err := rpcClient.Eth().GetBalance(
+				web3.Address(testCase.staker),
+				web3.Latest,
+			)
+			assert.NoError(t, err)
+
+			var out string
+			if callErr := rpcClient.Call(
+				"stake_getStakedBalance",
+				&out,
+				web3.Address(testCase.staker),
+				web3.Latest.String(),
+			); callErr != nil && testCase.shouldSucceed {
```

Comment:
I guess this isn't needed?
```suggestion
			); callErr != nil {
```

Reply:
Leftover from the merge.
Removed 👍